rust_src/src/xfaces.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{
+    lisp::defsubr,
+    remacs_sys::{clear_face_cache, windows_or_buffers_changed},
+};
+use remacs_macros::lisp_fn;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+lazy_static! {
+    static ref FACE_CHANGE: AtomicBool = AtomicBool::new(false);
+}
+
+/// Clear face caches on all frames.
+/// Optional THOROUGHLY non-nil means try to free unused fonts, too.
+#[lisp_fn(name = "clear-face-cache")]
+pub fn rust_clear_face_cache(thoroughly: bool) {
+    unsafe {
+        clear_face_cache(thoroughly);
+    }
+    FACE_CHANGE.store(true, Ordering::SeqCst);
+    unsafe {
+        windows_or_buffers_changed = 53;
```

Comment:
`53` is an arbitrary value. If you look at the code `windows_or_buffers_changed` is set to a different value each time it is assigned. This allows one to debug where the value came from. The value is only checked for zero or non-zero.

Reply:
See this comment in xdisp.c.

```
  Mostly for historical reasons, windows_or_buffers_changed can also take
   other non-zero values.  In that case, the precise value doesn't matter (it
   encodes the cause of the setting but is only used for debugging purposes),
   and what it means is that we shouldn't pay attention to any `redisplay' bits
   and we should simply try and redisplay every window out there.
```

Reply:
Do we want to keep the `53` floating here or do we want to convert it to an enum that looks something like:

```rust
enum ChangeCode {
    ClearFaceCache = 53,
    ...
}
```

?

Reply:
The use case is a developer in a debugger who sees '53' in `windows_or_buffers_changed` so they know the most recent thing to set the value was `Fclear_face_cache`. The enum would not help here since the C code is still using an int.